source/Halibut/Diagnostics/ExceptionReturnedByHalibutProxyExtensionMethod.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.IO;
+using System.Net.Sockets;
+using Halibut.Exceptions;
+using Halibut.Transport;
+using Halibut.Transport.Protocol;
+using Halibut.Transport.Proxy.Exceptions;
+
+namespace Halibut.Diagnostics
+{
+
+    public static class ExceptionReturnedByHalibutProxyExtensionMethod
+    {
+        public static HalibutNetworkExceptionType IsNetworkError(this Exception exception)
+        {
+            if (exception is NoMatchingServiceOrMethodHalibutClientException)
+            {
+                return HalibutNetworkExceptionType.NotANetworkError;
+            }
+
+            if (IsErrorInService(exception))
+            {
+                return HalibutNetworkExceptionType.NotANetworkError;
+            }
+
+            if (exception is UnexpectedCertificateException
+                || exception is FileNotFoundException)
+            {
+                return HalibutNetworkExceptionType.NotANetworkError;
+            }
+            
+            
+
+            if (exception is ProtocolException
+                || exception is SocketException
+                || exception is IOException)
+            {
+                return HalibutNetworkExceptionType.IsNetworkError;
+            }
+
+            if (exception is ProxyException)
+            {
+                if ((exception as ProxyException).CausedByNetworkError) return HalibutNetworkExceptionType.IsNetworkError;
+                return HalibutNetworkExceptionType.NotANetworkError;
+            }
+            
+            if (exception is HalibutClientException && exception.Message.Contains("System.IO.FileNotFoundException"))
+            {
+                return HalibutNetworkExceptionType.NotANetworkError;
+            }
+            
+            if (exception is HalibutClientException && exception.InnerException != null)
+            {
+                return IsNetworkError(exception.InnerException);
+            }
+
+            
+
+            return HalibutNetworkExceptionType.UnknownError;
+        }
+
+        /// <summary>
+        /// Did the exception thrown from a HalibutProxy object one which was caused by the service e.g. Tentacle
+        /// throwing an exception?
+        /// </summary>
+        /// <param name="e"></param>
+        /// <returns></returns>
```

Comment:
Nit: incomplete doco. Also, does this need to be public?

Reply:
I think it would be handy to be able to identify tentacle errors, currently we do NOT log tentacle errors since we think they could contain sensitive information.